mne/io/edf/tests/test_edf.py

```diff
@@ -295,4 +295,44 @@ def test_to_data_frame():
         assert_array_equal(df.values[:, 0], raw._data[0] * 1e13)
 
 
+def test_read_annot(tmpdir):
+    """Test parsing the tal channel."""
+    EXPECTED_ANNOTATIONS = [[180.0, 0, 'Lights off'], [180.0, 0, 'Close door'],
+                            [180.0, 0, 'Lights off'], [180.0, 0, 'Close door'],
+                            [3.14, 4.2, 'nothing'], [1800.2, 25.5, 'Apnea']]
+    SFREQ = 100
+    DATA_LENGTH = int(EXPECTED_ANNOTATIONS[-1][0] * SFREQ) + 1
+    annot = (b'+180\x14Lights off\x14Close door\x14\x00\x00\x00\x00\x00'
+             b'+180\x14Lights off\x14\x00\x00\x00\x00\x00\x00\x00\x00'
+             b'+180\x14Close door\x14\x00\x00\x00\x00\x00\x00\x00\x00'
+             b'+3.14\x1504.20\x14nothing\x14\x00\x00\x00\x00'
+             b'+1800.2\x1525.5\x14Apnea\x14\x00\x00\x00\x00\x00\x00\x00'
+             b'+123\x14\x14\x00\x00\x00\x00\x00\x00\x00')
+    annot_file = tmpdir.join('annotations.txt')
+    annot_file.write(annot)
+    annotmap_file = tmpdir.join('annotations_map.txt')
+    annotmap_file.write('Lights off:1,nothing:2,Apnea:3,Close door:4')
+
+    stim_ch = _read_annot(annot=str(annot_file), annotmap=str(annotmap_file),
+                          sfreq=SFREQ, data_length=DATA_LENGTH)
+
+    assert stim_ch.shape == (DATA_LENGTH,)
+    assert_array_equal(np.bincount(stim_ch), [180018, 0, 1, 1, 1])
+
+
+def test_read_raw_edf_deprecation_of_annot_annotmap(tmpdir):
+    """Test deprecation of annot and annotmap."""
+    annot = (b'+0.1344\x150.2560\x14two\x14\x00\x00\x00\x00'
+             b'+0.3904\x151.0\x14two\x14\x00\x00\x00\x00'
+             b'+2.0\x14three\x14\x00\x00\x00\x00\x00\x00\x00\x00'
+             b'+2.5\x152.5\x14two\x14\x00\x00\x00\x00')
+    annot_file = tmpdir.join('annotations.txt')
+    annot_file.write(annot)
+    annotmap_file = tmpdir.join('annotations_map.txt')
+    annotmap_file.write('two:2,three:3')
+
+    with pytest.warns(DeprecationWarning):
```

Comment:
Why not pytest.depecated_call? It's the more standard way to deal with deprecated calls I think

Reply:
I'm not sure. 

```sh
~/code/mne-python _read_annot*
(mne27) ❯ git grep "pytest.depre"       
mne/tests/test_morph.py:    with pytest.deprecated_call():
mne/tests/test_morph.py:    with pytest.deprecated_call():
mne/tests/test_morph.py:    with pytest.deprecated_call():
mne/tests/test_utils.py:    pytest.deprecated_call(deprecated_func)
mne/tests/test_utils.py:    pytest.deprecated_call(deprecated_class)

~/code/mne-python _read_annot*
(mne27) ❯ git grep "pytest.warns(DeprecationWarning"
mne/io/brainvision/tests/test_brainvision.py:    with pytest.warns(DeprecationWarning):
mne/io/brainvision/tests/test_brainvision.py:    with pytest.warns(DeprecationWarning):
mne/io/edf/tests/test_edf.py:    with pytest.warns(DeprecationWarning):
mne/io/eeglab/tests/test_eeglab.py:    with pytest.warns(DeprecationWarning, match="read_events_eeglab"):
mne/io/eeglab/tests/test_eeglab.py:    with pytest.warns(DeprecationWarning, match="read_events_eeglab"):
mne/io/eeglab/tests/test_eeglab.py:    with pytest.warns(DeprecationWarning, match="read_events_eeglab"):
mne/io/eeglab/tests/test_eeglab.py:    with pytest.warns(DeprecationWarning, match="read_events_eeglab"):
mne/io/eeglab/tests/test_eeglab.py:    with pytest.warns(DeprecationWarning, match="read_events_eeglab"):
mne/io/eeglab/tests/test_eeglab.py:    with pytest.warns(DeprecationWarning, match="read_events_eeglab"):
mne/io/tests/test_raw.py:    with pytest.warns(DeprecationWarning, match='by assignment is deprecated'):
mne/io/tests/test_raw.py:    with pytest.warns(DeprecationWarning):
mne/io/tests/test_raw.py:    with pytest.warns(DeprecationWarning):
```

I like `with pytest.warns(DeprecationWarning, match="annot.*annotmap.*"):` better 'cos it makes explicit which deprecation warning is the target. (I agree, that I forgot here).

On the other side, I've gone back to pytest documentation https://docs.pytest.org/en/latest/warnings.html and they recommend `pytest.deprecated_call` when depreation warnings are filtered out in the setup (which is not our case).

I've also checked our website and there's no recommendation https://mne-tools.github.io/stable/contributing.html?highlight=contribute 

I say to enter this PR with `pytest.warns` + `match=blabla`, since most of the file is like this. And unify + document a decision basd on #5599